### PR TITLE
Remove legacy issuer credential

### PR DESCRIPTION
**Pull request by mirgee**
Signed-off-by: Miroslav Kovar <miroslavkovar@protonmail.com>

### CODE REVIEW

File: libvcx/src/issuer_credential.rs [hunk 1]
```diff
@@ -1,610 +1,23 @@
...
-use utils::libindy::payments::PaymentTxn;
-use utils::qualifier;
 use v3::handlers::issuance::Issuer;
 
 lazy_static! {
-    static ref ISSUER_CREDENTIAL_MAP: ObjectCache<IssuerCredentials> = ObjectCache::<IssuerCredentials>::new("issuer-credentials-cache");
+    static ref ISSUER_CREDENTIAL_MAP: ObjectCache<Issuer> = ObjectCache::<Issuer>::new("issuer-credentials-cache");
 }
 
+// This enum is left only to avoid making breaking serialization changes
```
Comment (Patrik-Stas) [inline libvcx/src/issuer_credential.rs:13]: In fact I think in the future we might accommodate same versioning strategy as was used before, having multiple version of state machines.

Comment (Patrik-Stas) [reply libvcx/src/issuer_credential.rs:13]: So I think it's fine if this stays

File: libvcx/src/messages/get_message.rs [hunk 2]
```diff
@@ -348,30 +343,22 @@ impl Message {
...
-            }
-            A2AMessage::CredentialOffer(offer) => {
-                if settings::is_strict_aries_protocol_set() {
-                    (PayloadKinds::CredOffer, json!(&offer).to_string())
-                } else {
-                    let cred_offer: CredentialOffer = offer.try_into()?;
-                    (PayloadKinds::CredOffer, json!(vec![cred_offer]).to_string())
-                }
+            A2AMessage::PresentationRequest(_) => {
+                convert_aries_message!(a2a_message, ProofRequest)
```
Comment (Patrik-Stas) [inline libvcx/src/messages/get_message.rs:354]: The macro doesn't seem to provide much utility anymore, I think we can just inline that code?

File: libvcx/src/api/issuer_credential.rs [hunk 3]
```diff
@@ -756,31 +709,8 @@ pub extern fn vcx_issuer_credential_get_payment_txn(command_handle: CommandHandl
...
-            }
-            Err(x) => {
-                error!("vcx_issuer_credential_get_payment_txn_cb(command_handle: {}, rc: {}, txn: {}) source_id: {}",
-                       command_handle, x, "null", source_id);
-                cb(command_handle, x.into(), ptr::null());
-            }
-        };
-
+        cb(command_handle, 1, ptr::null());
+        error!("Payments not supported yet");
```
Comment (Patrik-Stas) [inline libvcx/src/api/issuer_credential.rs:713]: `error!` should go before `cb`

File: libvcx/src/messages/get_message.rs (hunk 2 above)
Comment (mirgee) [reply libvcx/src/messages/get_message.rs:354]: True, will fix

File: libvcx/src/api/issuer_credential.rs (hunk 3 above)
Comment (mirgee) [reply libvcx/src/api/issuer_credential.rs:713]: Good catch

**Review by Patrik-Stas — APPROVED** [review]

**Pull request merged**